### DISCUSSION TITLE
Electric Field Calculation for Non-Brain Regions

### DISCUSSION THREAD

**Discussion by ianibis** [simnibs/simnibs#401] [9. Results and analysis] [answered]
Hello,
I tried to determine the electric field via TMS for regions adjacent to the brain, eg, the scalp, and found that the electric field did not differ from the cortical e-field value (at 4mm Skin Distance). For the scalp, after computing the e-field, my Gmsh pipeline was Tools>Visibility>Select Scalp Volume/Surface>Apply. I might be doing this wrong. Is there a way with the GUI or via MATLAB/Python to compute/visualize the e-fields of areas that aren't the WM or GM?
Thank you!

**Comment by oulap** [comment]
Hi,

I have not used gmsh for computing e-fields in tissues, but you can definitely do it in python. Here's an example of how to analyze the e-field in the gray matter:

https://github.com/simnibs/simnibs/blob/master/simnibs/examples/analysis/roi_analysis_mni.py

Note how the gray matter elements are extracted on line 18. If you want to use another tissue, just change GM to e.g., SCALP. The available tissue tags are listed here:

https://github.com/simnibs/simnibs/blob/d045e8ccda6b349cbec639563af32dac69e95909/simnibs/utils/mesh_element_properties.py#L5

**Reply by ianibis** [reply]
Thank you. I'll try that.

**Comment by TorgeW** [comment, marked as answer]
Your problem in gmsh was just that the min and max of the color bar are not updated automatically.
So to fix this go to Tools -> Options -> View [1] (or whatever view your magnE is) -> General -> Click on Max under Custom
That will set the color bar max to the maximum of the data you are viewing.

**Reply by ianibis** [reply]
Thank you for the response. Following these steps did change the maximum of the scalp field, but the max values do not always seem accurate, as shown in the Cool-DB80 coil image I attached (second image). Also, the visualization appears less smooth, as shown in both example images. I tried this with several coil types and locations, yet there were problems each time. Do you know of a solution to this issue?
<img width="425" alt="TMS_CoolB70Coil" src="https://github.com/user-attachments/assets/f9b8aac0-5206-4a8b-900f-ec448c80f8f6">
<img width="426" alt="TMS_CoolDB80Coil" src="https://github.com/user-attachments/assets/82f166b1-25ef-43a6-86fa-f6159d428581">

**Reply by TorgeW** [reply]
what you can see here are outliers, if you manually reduce the max value (and enable Saturate right next to it) you should get a better visualization. To me this looks like your coil is either intersecting with the head or extremely close. You can see the outliers are in a grid pattern in the first image which looks like the coil dipoles are inside the head.

**Reply by ianibis** [reply]
Ok, got it. Thanks again.
